src/ledgers/ledgers_request.rs

```diff
@@ -79,17 +58,38 @@ impl LedgersRequest {
     /// # Returns
     /// The request object
     /// [AllLedgersRequest](struct.AllLedgersRequest.html)
-    pub fn set_cursor(&mut self, cursor: u32) {
-        self.cursor = Some(cursor);
+    pub fn set_cursor(self, cursor: u32) -> Result<LedgersRequest, String>{
+        if cursor < 1 {
+            return Err("cursor must be greater than or equal to 1".to_string());
+        }
+
+        Ok(
+            LedgersRequest {
+                cursor: Some(cursor),
+                limit: self.limit,
+                order: self.order
+            }
+        )
     }
+
     /// Sets the limit
     /// # Arguments
     /// * `limit` - The limit
     /// # Returns
     /// The request object
     /// [AllAssetsRequest](struct.AllAssetsRequest.html)
-    pub fn set_limit(&mut self, limit: u32) {
-        self.limit = Some(limit);
+    pub fn set_limit(self, limit: u32) -> Result<LedgersRequest, String> {
+        if limit < 1 || limit > 200 {
+            return Err("limit must be between 1 and 200".to_string());
+        }
```

Comment:
The same remarks as ealier. These are still runtime exceptions, but could return an `InvalidLedgerRequest` state. 
Please update the docs indicating what the boundaries are for the cursor.
Also, let's size limit down to a u8.

Reply:
Moved to https://github.com/Baseflow/stellar-rust-sdk/issues/4 and https://github.com/Baseflow/stellar-rust-sdk/issues/8